pypowsybl/dynamic.py

```diff
@@ -0,0 +1,190 @@
+
+
+class EventMapping:
+    def __init__(self) -> None:
+        self._handle = _pp.create_event_mapping()
+
+    def add_quadripole_disconnection(self, event_model_id: str, static_id: str, parameter_set_id: str) -> None:
+        _pp.add_event_quadripole_disconnection(
+            self._handle, event_model_id, static_id, parameter_set_id)
+
+    def add_set_point_boolean(self, event_model_id: str, static_id: str, parameter_set_id: str) -> None:
+        _pp.add_event_set_point_boolean(
+            self._handle, event_model_id, static_id, parameter_set_id)
+
+    def add_event(self, parameter_set_id: str, event: EventType, static_id: str, event_id: str = None) -> None:
+        if not event_id:
+            event_id = str(uuid4())
+
+        if event is EventType.QUADRIPOLE_DISCONNECT:
+            _pp.add_event_quadripole_disconnection(
+                self._handle, event_id, static_id, parameter_set_id)
+            return
+        if event is EventType.SET_POINT_BOOLEAN:
+            _pp.add_event_set_point_boolean(
+                self._handle, event_id, static_id, parameter_set_id)
+            return
+
+        raise Exception(
+            f"Pypowsybl-DynamicSimulationError: Unknown event {event}")
+
+    @staticmethod
+    def get_possible_events() -> _List[EventType]:
+        return list(EventType)
+
+
+class SimulationResult:
+    def __init__(self, handle: _pp.JavaHandle) -> None:
+        self._handle = handle
+
+    def state(self) -> str:
+        return _pp.get_dynamic_simulation_results_status(self._handle)
+
+    @property
+    def status(self) -> str:
+        return self.state()
+
+    def get_curve(self, curve_name: str) -> _pd.DataFrame:
+        series_array = _pp.get_dynamic_curve(self._handle, curve_name)
+        return create_data_frame_from_series_array(series_array)
+
+    def get_all_curves(self, curve_mapping: CurveMapping, dynamic_id: str) -> _pd.DataFrame:
+        curve_name_lst = _pp.get_all_dynamic_curves_ids(
+            curve_mapping._handle, dynamic_id)
+        df_curves = [self.get_curve(curve_name)
+                     for curve_name in curve_name_lst]
+        return _pd.concat(df_curves, axis=1)
+
+
+class Simulation:
+    def __init__(self) -> None:
+        self._handle = _pp.create_dynamic_simulation_context()
+
+    def run(self,
+            network: _Network,
+            model_mapping: ModelMapping,
+            event_mapping: EventMapping,
+            timeseries_mapping: CurveMapping,
+            start: int,
+            stop: int,
+            ) -> SimulationResult:
+        return SimulationResult(
+            _pp.run_dynamic_model(
+                self._handle,
+                network._handle,
+                model_mapping._handle,
+                event_mapping._handle,
+                timeseries_mapping._handle,
+                start, stop)
+        )
+
+    @staticmethod
+    def set_config(path: str, config_name: str) -> None:
```

Comment:
Should be removed until we support changing the config at runtime.

Reply:
You are right, it was not meant ot be pushed